pyopenms tests failing on macOS

Maybe related to RPATH and fix_dependency script changes in #6421

Looks like the latest updates didn't fix it.
`
cp: @rpath/QtCore.framework/Versions/5/QtCore: No such file or directory
chmod: /Users/runner/work/OpenMS/OpenMS/bld/pyOpenMS/pyopenms/QtCore: No such file or directory
error: /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file: @rpath/QtCore.framework/Versions/5/QtCore (No such file or directory)
Exception `NoMethodError' at ./mac_fix_dependencies.rb:94 - undefined method `strip' for nil:NilClass
./mac_fix_dependencies.rb:94:in `extractInstallName': undefined method `strip' for nil:NilClass (NoMethodError)
	from ./mac_fix_dependencies.rb:265:in `handleFramework'
	from ./mac_fix_dependencies.rb:147:in `block in handleDependencies'
	from ./mac_fix_dependencies.rb:122:in `each'
	from ./mac_fix_dependencies.rb:122:in `handleDependencies'
	from ./mac_fix_dependencies.rb:272:in `handleFramework'
	from ./mac_fix_dependencies.rb:147:in `block in handleDependencies'
	from ./mac_fix_dependencies.rb:122:in `each'
	from ./mac_fix_dependencies.rb:122:in `handleDependencies'
	from ./mac_fix_dependencies.rb:322:in `handleDyLib'
	from ./mac_fix_dependencies.rb:149:in `block in handleDependencies'
	from ./mac_fix_dependencies.rb:122:in `each'
	from ./mac_fix_dependencies.rb:122:in `handleDependencies'
	from ./mac_fix_dependencies.rb:322:in `handleDyLib'
	from ./mac_fix_dependencies.rb:440:in `block (2 levels) in <main>'
	from ./mac_fix_dependencies.rb:433:in `each'
	from ./mac_fix_dependencies.rb:433:in `block in <main>'
	from ./mac_fix_dependencies.rb:431:in `chdir'
	from ./mac_fix_dependencies.rb:431:in `<main>'
make[3]: *** [src/pyOpenMS/CMakeFiles/pyopenms] Error 1
make[2]: *** [src/pyOpenMS/CMakeFiles/pyopenms.dir/all] Error 2
make[1]: *** [src/pyOpenMS/CMakeFiles/pyopenms.dir/rule] Error 2
make: *** [pyopenms] Error 2
Error: Process completed with exit code 2.
`

--- a/cmake/MacOSX/fix_dependencies.rb
+++ b/cmake/MacOSX/fix_dependencies.rb
@@ -294,7 +294,7 @@ def handleFramework(frameworkPath, targetPath, rpaths)
       end
   else
     debug "RPath but no lib found. Assuming fixed already: #{libname}"
-    if $EXTRACTFW
+    if !$EXTRACTFW
       # But at least adjust the libname which is used in handleDependencies to fix loading.
       # get framework path (name of dir containing .framework)
       frameworkDir=frameworkPath.to_s.gsub(/(.*\.framework).*/,'\1')

--- a/cmake/MacOSX/fix_dependencies.rb
+++ b/cmake/MacOSX/fix_dependencies.rb
@@ -294,14 +294,18 @@ def handleFramework(frameworkPath, targetPath, rpaths)
       end
   else
     debug "RPath but no lib found. Assuming fixed already: #{libname}"
-    # But at least adjust the libname which is used in handleDependencies to fix loading.
-    # get framework path (name of dir containing .framework)
-    frameworkDir=frameworkPath.to_s.gsub(/(.*\.framework).*/,'\1')
-    frameworkName=frameworkDir.to_s.gsub(/(.*\/)([^\/]*\.framework)/,'\2')
+    if $EXTRACTFW
+      # But at least adjust the libname which is used in handleDependencies to fix loading.
+      # get framework path (name of dir containing .framework)
+      frameworkDir=frameworkPath.to_s.gsub(/(.*\.framework).*/,'\1')
+      frameworkName=frameworkDir.to_s.gsub(/(.*\/)([^\/]*\.framework)/,'\2')
 
-    # the actual lib name
-    internFrameworkDir=frameworkPath.to_s.gsub(/(.*\.framework)\/(.*)/,'\2')
-    libname=frameworkName+'/'+internFrameworkDir
+      # the actual lib name
+      internFrameworkDir=frameworkPath.to_s.gsub(/(.*\.framework)\/(.*)/,'\2')
+      libname=frameworkName+'/'+internFrameworkDir
+    else
+      libname=File.basename(frameworkPath)
+    end
     # the new path
     newFrameworkPath="#{targetPath}/#{libname}"
   end

